fix(stdlib): io test

notJoon opened the pull request:
# Description

Fixed a failing test in the io package. Additionally, I activated some previously commented-out tests by bypassing the need for `os.CreateTemp` using `bytes.Buffer`.

<details><summary>Contributors' checklist...</summary>

- [ ] Added new tests, or not needed, or not feasible
- [ ] Provided an example (e.g. screenshot) to aid review or the PR is self-explanatory
- [ ] Updated the official documentation or not needed
- [ ] No breaking changes were made, or a `BREAKING CHANGE: xxx` message was included in the description
- [ ] Added references to related issues and PRs
- [ ] Provided any useful hints for running manual tests
- [ ] Added new benchmarks to [generated graphs](https://gnoland.github.io/benchmarks), if any. More info [here](https://github.com/gnolang/gno/blob/master/.benchmarks/README.md).
</details>

thehowl reviewed (changes requested):
Instead of exporting the errors, I'd rather we kept them unexported and put the test in `package io` (instead of io_test).

notJoon commented:
> Instead of exporting the errors, I'd rather we kept them unexported and put the test in `package io` (instead of io_test).

Changed the error types to private and modified the test package declaration from `io_test` to `io`.

The `ErrShortWrite`, `EOF`, `ErrUnexpectedEOF`, and `ErrNoProgress` errors remain public, as they are also used in the `bufio` package.

https://github.com/gnolang/gno/pull/2898/commits/a63bb39a4285b8c14a8586bf27990028165e9e93
